service/actions/websocket.js

```diff
@@ -0,0 +1,460 @@
+    }
+}
+
+function handleUserUpdatedEvent(msg, dispatch, getState) {
+    const entities = getState().entities;
+    const users = entities.users;
+    const user = msg.data.user;
+
+    if (user.id !== users.currentId) {
+        dispatch({
+            type: UsersTypes.RECEIVED_PROFILES,
+            data: {
+                [user.id]: user
+            }
+        }, getState);
+    }
+}
+
+function handleChannelViewedEvent(msg, dispatch, getState) {
+    const state = getState();
+    const channels = state.entities.channels;
+    const teams = state.entities.teams;
+    const users = state.entities.users;
+
+    if (teams.currentId === msg.broadcast.team_id && channels.currentId !== msg.data.channel_id &&
+        users.currentId === msg.broadcast.user_id) {
+        getChannelHelper(teams.currentId, msg.data.channel_id, dispatch, getState);
+    }
+}
+
+function handleChannelDeletedEvent(msg, dispatch, getState) {
+    const state = getState();
+    const channels = state.entities.channels;
+    const teams = state.entities.teams;
+
+    if (msg.data.channel_id === channels.currentId) {
+        let channelId = '';
+        const channel = Object.keys(channels.channels).filter((key) => channels.channels[key].name === Constants.DEFAULT_CHANNEL);
+
+        if (channel.length) {
+            channelId = channel[0];
+        }
+
+        dispatch({type: ChannelTypes.SELECTED_CHANNEL, data: channelId}, getState);
+    }
+
+    if (msg.broadcast.team_id === teams.currentId) {
+        dispatch({type: ChannelTypes.RECEIVED_CHANNEL_DELETED, data: msg.data.channel_id}, getState);
+        getChannelsAndMembersHelper(teams.currentId, dispatch, getState);
+    }
+}
+
+function handleDirectAddedEvent(msg, dispatch, getState) {
+    const state = getState();
+    const teams = state.entities.teams;
+
+    getChannelHelper(teams.currentId, msg.broadcast.channel_id, dispatch, getState);
+}
+
+function handlePreferenceChangedEvent(msg, dispatch, getState) {
+    const preference = JSON.parse(msg.data.preference);
+    dispatch({type: PreferencesTypes.RECEIVED_PREFERENCES, data: [preference]}, getState);
+
+    if (preference.category === Constants.CATEGORY_DIRECT_CHANNEL_SHOW) {
+        const state = getState();
+        const users = state.entities.users;
+        const userId = preference.name;
+        const status = users.statuses[userId];
+
+        if (!users.profiles[userId]) {
+            getProfilesHelper([userId], dispatch, getState);
+        }
+
+        if (status !== Constants.ONLINE) {
+            getStatusesHelper([userId], dispatch, getState);
+        }
+    }
+}
+
+function handleStatusChangedEvent(msg, dispatch, getState) {
+    dispatch({
+        type: UsersTypes.RECEIVED_STATUSES,
+        data: {
+            [msg.data.user_id]: msg.data.status
+        }
+    }, getState);
+}
+
+// Helpers
+
+function getChannelsAndMembersHelper(teamId, dispatch, getState) {
+    Client.getChannels(teamId).then((data) => {
```

Comment:
Like I mentioned above, I think we should be dispatching service actions for all of these instead of talking directly to the client. That'll give us request statuses, make global error state easier to track, and reduce some extra code that's being called here. Thoughts?